### PR TITLE
smoking IN SPACE

### PR DESCRIPTION
**Pull request by SonixApache**
lets you keep your cigarettes lit in no-oxygen environments, as long as your internals tank is hooked up and has oxygen in it - just like long long ago
does not however consume oxygen
tested

🆑 
 - tweak: You can now use lit cigarettes as internals again, without them fizzling out when there's no air!

### CODE REVIEW

**Comment by Exxion** [comment]
Only if you sprite a tube connecting to the end of it

**Comment by SonixApache** [comment]
suspension of disbelief 🙂

**Comment by Lokiamis** [comment]
This is stupid

File: code/game/objects/items/weapons/cigs_lighters.dm [hunk 1]
```diff
@@ -356,6 +356,10 @@ MATCHBOXES ARE ALSO IN FANCY.DM
 	smoketime--
 	var/datum/gas_mixture/env = location.return_air()
 	if(smoketime <= 0 | env.molar_density(GAS_OXYGEN) < (5 / CELL_VOLUME))
```
Comment (DamianX) [inline code/game/objects/items/weapons/cigs_lighters.dm]: 🤔

Comment (SonixApache) [reply code/game/objects/items/weapons/cigs_lighters.dm]: 🤔

File: code/game/objects/items/weapons/cigs_lighters.dm [hunk 2]
```diff
@@ -356,6 +356,10 @@ MATCHBOXES ARE ALSO IN FANCY.DM
...
+		if(smoketime > 0 && ishuman(loc))
+			var/mob/living/carbon/human/mysmoker = loc
+			if(mysmoker.internal && mysmoker.internal.air_contents.partial_pressure(GAS_OXYGEN) > 0)
```
Comment (Exxion) [inline code/game/objects/items/weapons/cigs_lighters.dm]:
```suggestion
			if(mysmoker.internal?.air_contents.partial_pressure(GAS_OXYGEN) > 0)
```
We live in the age of 512 now

File: code/game/objects/items/weapons/cigs_lighters.dm (hunk 1 above)
Comment (Exxion) [reply code/game/objects/items/weapons/cigs_lighters.dm]: Apparently the `|` has lower precedence than the comparisons for some fucking reason, resulting in it being equivalent to what it *should* be written as

File: code/game/objects/items/weapons/cigs_lighters.dm (hunk 2 above)
Comment (SonixApache) [reply code/game/objects/items/weapons/cigs_lighters.dm]: magic

File: code/game/objects/items/weapons/cigs_lighters.dm (hunk 1 above)
Comment (SonixApache) [reply code/game/objects/items/weapons/cigs_lighters.dm]: i am unable to decipher your message and thus my answer remains
🤔

Comment (Exxion) [reply code/game/objects/items/weapons/cigs_lighters.dm]: `|` `!=` `||`

Comment (SonixApache) [reply code/game/objects/items/weapons/cigs_lighters.dm]: so replace the `|` with a `||`, got it

Comment (ShiftyRail) [reply code/game/objects/items/weapons/cigs_lighters.dm]: In simple terms,
`(smoketime <= 0 | env.molar_density(GAS_OXYGEN)` first does "`smoketime <= 0`", returns 0 or 1, and then it bitwise OR (which is equivalent to regular OR as we have only 1 bit to go by

This is equivalent to what it is intended to do, but by pure mistake. Changing it to `||` will help readability and will look less stupid

**Comment by cowboybuttsdrivemenuts** [comment]
merge when? not being able to smoke in my ENCLOSED space helmet is annoying

**Pull request merged**